### PR TITLE
[CLOUD-4107] - JDK 17 images

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.redhat.com/browse/CLOUD-4107
Signed-off-by: Ken Wills <kwills@redhat.com>

### CODE REVIEW

File: jdk17-image/image.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+modules:
+      repositories:
+          - name: cct_module
+            git:
+                  url: https://github.com/jboss-openshift/cct_module.git
+                  ref: 0.45.4
+          - name: jboss-eap-modules
+            git:
+                  url: https://github.com/jboss-container-images/jboss-eap-modules.git
+                  ref: EAP_745_CR1_JDK11
```
Comment (jmesnil) [inline jdk17-image/image.yaml:49]: is that correct? Do the JDK 17 images depend on a tag `EAP_745_CR1_JDK11`?

File: jdk17-image/image.yaml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    - name: "io.openshift.tags"
+      value: "builder,javaee,eap,eap7"
+    - name: "io.openshift.s2i.scripts-url"
+      value: "image:///usr/local/s2i"
+    - name: "maintainer"
+      value: "Red Hat"
+envs:
+      - name: DISABLE_LEGACY_SECURITY
+        value: "true"
+      - name: AB_JOLOKIA_OFF
```
Comment (jfdenise) [inline jdk17-image/image.yaml]: It seems from discussions that we are not disabling Jolokia for this release.

File: runtime-image/jdk17-overrides.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+from: "registry.redhat.io/ubi8/ubi:latest"
+
+labels:
+    - name: "com.redhat.component"
+      value: "jboss-eap-74-openjdk17-runtime-openshift-rhel8-container"
+
+envs:
+     - name: DISABLE_LEGACY_SECURITY
+       value: "true"
+     - name: AB_JOLOKIA_OFF
```
Comment (jfdenise) [inline runtime-image/jdk17-overrides.yaml]: It seems from discussions that we are not disabling Jolokia for this release.

**Review by jfdenise — CHANGES_REQUESTED** [review]

File: jdk17-image/image.yaml (hunk 1 above)
Comment (luck3y) [reply jdk17-image/image.yaml:47]: +1, updated

File: jdk17-image/image.yaml (hunk 2 above)
Comment (luck3y) [reply jdk17-image/image.yaml]: +1 updated

File: runtime-image/jdk17-overrides.yaml (hunk 3 above)
Comment (luck3y) [reply runtime-image/jdk17-overrides.yaml]: +1 updated.

File: jdk17-image/image.yaml [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+modules:
+      repositories:
+          - name: cct_module
+            git:
+                  url: https://github.com/jboss-openshift/cct_module.git
+                  ref: 0.45.4
+          - name: jboss-eap-modules
+            git:
+                  url: https://github.com/jboss-container-images/jboss-eap-modules.git
+                  ref: EAP_745_CR1_JDK11
```
Comment (jmesnil) [inline jdk17-image/image.yaml]: I don't see this tag in  https://github.com/jboss-container-images/jboss-eap-modules/tags. There is only a `EAP_745_CR1`. Besides, is it normal to use a `_JDK11` tag to build the JDK17 images?

**Review by jmesnil — COMMENTED** [review]
Looks good to me aside of the missing tag from https://github.com/jboss-container-images/jboss-eap-modules.git

File: jdk17-image/image.yaml (hunk 4 above)
Comment (luck3y) [reply jdk17-image/image.yaml]: Ah, sorry, this was a typo, updated to JDK 17. 

The tag on jboss-eap-modules will be created after https://github.com/jboss-container-images/jboss-eap-modules/pull/264 is merged.

**Review by jfdenise — APPROVED** [review]

**Review by jmesnil — APPROVED** [review]

File: jdk17-image/image.yaml (hunk 4 above)
Comment (luck3y) [reply jdk17-image/image.yaml]: This is merged and tagged now.

**Pull request merged**